scripts/adage_pancancer.py

```diff
@@ -0,0 +1,115 @@
+scripts/adage_pancancer.py
+
+Comparing a VAE learned features to ADAGE features. Use this script within
+the context of a parameter sweep to compare performance across a grid of
+hyper parameters.
+
+Usage:
+
+    Run in command line with required command arguments:
+
+        python scripts/adage_pancancer.py --learning_rate
+                                          --batch_size
+                                          --epochs
+                                          --sparsity
+                                          --noise
+                                          --output_filename
+
+    Typically, arguments to this script are compiled automatically by:
+
+        python scripts/vae_paramsweep.py --parameter_file <parameter-filepath>
+                                         --config_file <configuration-filepath>
+
+Output:
+Loss and validation loss for the specific model trained
+"""
+
+import os
+import argparse
+import numpy as np
+import pandas as pd
+
+from keras.layers import Input, Dense
+from keras.layers.noise import GaussianDropout
+from keras.models import Model
+from keras import optimizers
+from keras.regularizers import l1
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-l', '--learning_rate',
+                    help='learning rate of the optimizer')
+parser.add_argument('-b', '--batch_size',
+                    help='Number of samples to include in each learning batch')
+parser.add_argument('-e', '--epochs',
+                    help='How many times to cycle through the full dataset')
+parser.add_argument('-s', '--sparsity',
+                    help='How much L1 regularization penalty to apply')
+parser.add_argument('-n', '--noise',
+                    help='How much Gaussian noise to add during training')
+parser.add_argument('-f', '--output_filename',
+                    help='The name of the file to store results')
+args = parser.parse_args()
+
+# Set hyper parameters
+learning_rate = float(args.learning_rate)
+batch_size = int(args.batch_size)
+epochs = int(args.epochs)
+sparsity = float(args.sparsity)
+noise = float(args.noise)
+output_filename = args.output_filename
+
+original_dim = 5000
+latent_dim = 100
+
+# Random seed
+seed = int(np.random.randint(low=0, high=10000, size=1))
+np.random.seed(seed)
+
+# Load Data
+rnaseq_file = os.path.join('data', 'pancan_scaled_zeroone_rnaseq.tsv')
+rnaseq_df = pd.read_table(rnaseq_file, index_col=0)
+
+# Split 10% test set randomly
+test_set_percent = 0.1
+rnaseq_test_df = rnaseq_df.sample(frac=test_set_percent)
+rnaseq_train_df = rnaseq_df.drop(rnaseq_test_df.index)
+
+# Input place holder for RNAseq data with specific input size
+input_rnaseq = Input(shape=(original_dim, ))
+encoded_rnaseq = GaussianDropout(noise)(input_rnaseq)
+encoded_rnaseq_2 = Dense(latent_dim, activation='relu',
+                         activity_regularizer=l1(sparsity))(encoded_rnaseq)
+decoded_rnaseq = Dense(original_dim, activation='relu')(encoded_rnaseq_2)
```

Comment:
Thanks for bringing this up - in initial development I tried several, all with varying results (Ideally I would have been documenting this, I know...) but after some quick prototyping I saw that `relu` had lower loss than both `sigmoid` and `tanh`.

However, I didn't look at downstream consequences of the model, including reconstruction. An issue I was having with `relu` is that some genes were simply not being reconstructed.

![relu_gene_reconstruction](https://user-images.githubusercontent.com/7340421/29790560-4ce43364-8c08-11e7-9d3a-fdbcdfd62722.png)

However, with `sigmoid` we no longer had this issue.

![sigmoid_gene_reconstruction](https://user-images.githubusercontent.com/7340421/29790252-2ebbb9f8-8c07-11e7-89ef-79f4a0180ec0.png)

Both models were trained with these params - which are different than my initial testing.

| param | optimal |
| :------: | :--------: |
| num_features | 5000 |
| encoding_dim | 100 |
| sparsity | 1e-6 |
| noise | 0.1 |
| epochs | 80 |
| batch_size | 50 |
| learning_rate | 0.9 |

Reply:
Moral of the story is that I need to rerun the parameter sweep with a `sigmoid` loss.

Before I get this merged, I also need to investigate:

1. Is [`GaussianDropout` ](https://github.com/fchollet/keras/blob/master/keras/layers/noise.py#L52) doing what I think its doing
2. Is the `encoder` model and `decoder` model accurate? I may elect to simply use the weights learned by the autoencoder to obtain a latent representation instead of `encoder.predict()`. I think there may be something buggy going on here too.